### PR TITLE
add doc about commitmetadata

### PR DESCRIPTION
**Pull request by CodingCat**
followup PR of https://github.com/apache/iceberg/commit/893af4a19841ae23e18b1e2196df9176d9d90bc2

adding doc about org.apache.iceberg.spark.CommitMetadata

### CODE REVIEW

**Comment by CodingCat** [comment]
Hi, @flyrain , just filed the followup PR here

File: docs/spark-configuration.md [hunk 1]
```diff
@@ -194,3 +194,17 @@ df.write
 | check-ordering       | true        | Checks if input schema and table schema are same  |
 | isolation-level | null | Desired isolation level for Dataframe overwrite operations.  `null` => no checks (for idempotent writes), `serializable` => check for concurrent inserts or deletes in destination partitions, `snapshot` => checks for concurrent deletes in destination partitions. |
 | validate-from-snapshot-id | null | If isolation level is set, id of base snapshot from which to check concurrent write conflicts into a table. Should be the snapshot before any reads from the table. Can be obtained via [Table API](../../api#table-metadata) or [Snapshots table](../spark-queries#snapshots). If null, the table's oldest known snapshot is used. |
+
+specifically, if you run SQL statements, you could use `org.apache.iceberg.spark.CommitMetadata` to add entries with custom-keys and corresponding values in the snapshot summary
```
Comment (flyrain) [inline docs/spark-configuration.md:198]: Minor suggestion:
`CommitMetadata` provides an interface to add custom metadata to a snapshot summary during a SQL execution, which can be beneficial for purposes such as auditing or change tracking. Here is an example:

File: docs/spark-configuration.md [hunk 2]
```diff
@@ -194,3 +194,17 @@ df.write
...
+
+```java
+import org.apache.iceberg.spark.CommitMetadata;
```
Comment (flyrain) [inline docs/spark-configuration.md:201]: Style suggestion: add an empty line

File: docs/spark-configuration.md [hunk 3]
```diff
@@ -194,3 +194,17 @@ df.write
...
+Map<String, String> properties = Maps.newHashMap();
+properties.put("property_key", "property_value");
+CommitMetadata.withCommitProperties(properties,
+        () -> {
+        spark.sql("DELETE FROM " + tableName + " where id = 1");
+        return 0;
```
Comment (flyrain) [inline docs/spark-configuration.md:207]: Indentation?

**Review by flyrain — COMMENTED** [review]
LGTM. Thanks @CodingCat. Left minor comments. Sorry for the delay.

**Comment by CodingCat** [comment]
@flyrain sorry for the delay, just addressed the comments, please take another look

File: docs/spark-configuration.md (hunk 2 above)
Comment (flyrain) [reply docs/spark-configuration.md:202]: Sorry, I was suggestion an empty under the `import` line.  Minor thing though.

**Review by flyrain — APPROVED** [review]
LGTM. Thanks @CodingCat !

File: docs/spark-configuration.md (hunk 2 above)
Comment (CodingCat) [reply docs/spark-configuration.md:201]: sure, updated

**Comment by flyrain** [comment]
Merged, thanks @CodingCat !

**Pull request merged**